src/wallet/mod.rs

```diff
@@ -1092,11 +1097,23 @@ where
             .iter_mut()
             .zip(psbt.global.unsigned_tx.input.iter())
         {
-            let prev_script = match prev_script_pubkeys.get(&input.previous_output) {
-                Some(prev_script) => prev_script,
+            let utxo = match lookup_output.get(&input.previous_output) {
+                Some(utxo) => utxo,
                 None => continue,
             };
 
+            let is_segwit = utxo.txout.script_pubkey.is_witness_program();
```

Comment:
I'm not entirely sure about this, I think nested witness scripts here would be `false`, because the `script_pubkey` looks like a normal P2SH

Reply:
Very good catch. I agree this is wrong. I am not sure if current behavior is right (see comment below). If we can figure it out I'll write tests to check for this.

Reply:
I've just removed this bit for now and added a test to make sure we don't accidentally introduce the problem in the future.